### ISSUE TITLE
Overuse of exact matches in eval.

### ISSUE THREAD

**Issue by afourney** [web-arena-x/webarena#134]
I would like to re-open issue #104 

There's an overuse of exact matches in the eval harness. For example, consider task 649:

```
 "intent": "Post in history subreddit about what could diffusion model help the correpong field.",
...
          "required_contents": {
            "must_include": [
              "diffusion model",
              "help"
            ]
          }
      
Our agent created the following post, but failed the test because it does not include the word "help":
![image](https://github.com/web-arena-x/webarena/assets/4017093/6bca11d9-3d8c-4371-8ef5-af0a74414edd)

This is overly rigid, and it is not an outlier.

**Comment by afourney** [comment]
@deepak-akkil

**Comment by afourney** [comment]
Here's another example. Task 643.

```
    "intent": "Post a notice on a virtual meetup for racing cars enthusiasts on Oct 21st in the nyc subreddit",
          
    "required_contents": {
            "must_include": [
              "racing cars",
              "Oct 21st",
              "virtual meetup"
            ]
          }
```

Our agent posted:

![image](https://github.com/web-arena-x/webarena/assets/4017093/723c2324-92c0-4cb7-8ab8-cb10348b47ce)

Which I believe is failing because of the use of "Racing car" (singular), rather than "racing cars" (plural), and possibly also the use of "October 21st" rather than "Oct 21st" -- though I've not traced this failure to confirm.

**Comment by shuyanzhou** [comment]
Hi @afourney this is a great point! We attempted to minimize the number of LLM-based evaluations to keep the evaluation straightforward. However, due to the different prompts and implementations, the agents can behave differently, but all correctly. I am re-examing the annotations to see to what extent and how we want to update the verifiers. We also want to make sure the comparisons across works are fair.

**Comment by afourney** [comment]
Agreed that changing scoring function now would diverge from runs already completed. Perhaps some versioning of the benchmark would be key here. But I'm pretty convinced that addressing such issues will improve the quality and utility of WebArena.

**Comment by gagb** [comment]
Same issue with task 27 -- string match requires 0. Our agents said zero and it got a fail.

**Comment by gagb** [comment]
> Hi @afourney this is a great point! We attempted to minimize the number of LLM-based evaluations to keep the evaluation straightforward. However, due to the different prompts and implementations, the agents can behave differently, but all correctly. I am re-examing the annotations to see to what extent and how we want to update the verifiers. We also want to make sure the comparisons across works are fair.

I think we need to at least (vastly) expand the patterns used for exact match. Or get rid of exact match and only use LLMs. The cost of LLM evaluation seems to be lower than what agents typically use to solve the task so it does sound like a big deal. If noise in evaluation is a worry, then it isn't clear which is noisier -- exact match (because of low recall) or LLM based eval (because from stochasticity in generations).

**Comment by peterychang** [comment]
Task 266 has the same issue

```
    "intent": "What's the closest national park to the largest city in Maine?",
    "eval": {
      "eval_types": [
        "string_match"
      ],
      "reference_answers": {
        "exact_match": "Acadia National Park"
      },
```

Our agent responded with `Acadia National Park is the closest national park to Portland, Maine.`

**Comment by shuyanzhou** [comment]
Thanks for the excellent catches. We are working on releasing the new evaluation, which will address such false negative judgments.
